datafusion/functions-aggregate/src/correlation.rs

```diff
@@ -487,17 +487,14 @@ impl GroupsAccumulator for CorrelationGroupsAccumulator {
             if denominator == 0.0 {
                 // TODO: Evaluate as `Null` (see notes above)
                 values.push(0.0);
-                nulls.append(false);
+                nulls.append_null();
             } else {
                 values.push(numerator / denominator);
-                nulls.append(true);
+                nulls.append_non_null();
             }
         }
 
-        Ok(Arc::new(Float64Array::new(
-            values.into(),
-            Some(nulls.finish().into()),
-        )))
+        Ok(Arc::new(Float64Array::new(values.into(), nulls.finish())))
```

Comment:
❤️  -- it is great. Again this will avoid creating a NullBuffer if the input had no nulls